rustls/src/lib.rs

```diff
@@ -424,6 +465,16 @@
 #![cfg_attr(bench, feature(test))]
 #![no_std]
 
+// This constraint is also enforced by `portable-atomic` crate - enforcing here as well for extra clarity
+// (with a QUICK WORKAROUND)
+#[cfg(all(
+    feature = "critical-section",
+    feature = "unsafe-assume-single-core",
+    // QUICK WORKAROUND for issue with CI semver job:
+    not(feature = "std"),
+))]
+compile_error!("invalid combination of `critical-section` & `unsafe-assume-single-core` features");
```

Comment:
As I said in issue #27, I think it would be a nicer developer experience if this kind of feature combination could be supported - see issue #27 for details.

Of course, this would need an update to be proposed & accepted in `portable-atomic`.

I would also like to add a TODO comment with this info.